### PR TITLE
fix(astro): type generation for empty collections

### LINKED ISSUE #11075: Empty Content Collections with defined scema don't generate typescript types
### Astro Info

```block
Astro                    v4.8.4
Node                     v22.0.0
System                   Linux (x64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             @astrojs/tailwind
                         @astrojs/svelte
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

I want to publish a blank vesion of my website with no content. I already have a schema set up on config.ts, and doing a dev build complains but works no problem. However, running a full production builds throws many typescript errors (specifically that something's of a type unknown) because the type wasn't properly generated. The example I provided should be enough to get the bug to trigger, but I wasn't able to.

### What's the expected result?

In this scenario, I would expect a build with no pages, similar to how a dev build works

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-2n2i41

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by Fryuni**
## Changes

- Generate a placeholder type to allow for type-safety, intellisense and autocomplete on empty collections.
- Closes #11075

## Testing

Expected types added on tests for `astro sync`

## Docs

I don't think this needs any docs change

### CODE REVIEW

File: .changeset/light-bugs-shake.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes type generation for empty collections
```
Comment (florian-lefebvre) [inline .changeset/light-bugs-shake.md:5]:
```suggestion
Fixes type generation for empty content collections
```

Comment (Fryuni) [reply .changeset/light-bugs-shake.md:5]: Also applies for data collections

Comment (florian-lefebvre) [reply .changeset/light-bugs-shake.md:5]: True but I think the term "content collections" cover the usage of defineCollection basically. I mean iirc in the docs content collections is always used

File: packages/astro/src/content/types-generator.ts [hunk 2]
```diff
@@ -429,12 +429,17 @@ async function writeContentFiles({
 					collectionConfig?.type ?? 'data'
 				: collection.type;
 
+		const collectionEntryKeys = Object.keys(collection.entries).sort();
+		const dataType = collectionConfig?.schema ? `InferEntrySchema<${collectionKey}>` : 'any';
 		switch (resolvedType) {
 			case 'content':
+				if (collectionEntryKeys.length === 0) {
+					contentTypesStr += `${collectionKey}: Record<string, {\n  id: string;\n  slug: string;\n  body: string;\n  collection: ${collectionKey};\n  data: ${dataType};\n  render(): Render[".md"];\n}>;\n`;
```
[the remark this answers is not in the document]
Comment (Fryuni) [reply packages/astro/src/content/types-generator.ts:437]: Good point, I'll try to save some time to do a little refactor around that.

**Review by ematipico — APPROVED** [review]

**Pull request merged**